runtime/metricsview/executor.go

```diff
@@ -320,3 +322,150 @@ func (e *Executor) Export(ctx context.Context, qry *Query, executionTime *time.T
 		"args": args,
 	})
 }
+
+// SearchQuery executes the provided query against the metrics view.
+func (e *Executor) Search(ctx context.Context, qry *SearchQuery, executionTime *time.Time) ([]SearchResult, error) {
+	if !e.security.CanAccess() {
+		return nil, runtime.ErrForbidden
+	}
+
+	// Generate a metricsview.Query and build a AST
+	// This is a hacky implementation since both metricsview.Query and AST are designed for aggregate queries.
+	// TODO :: Refactor the code and extract common functionality from metricsview.Query and AST and write SearchQuery to underlying SQL/Native druid query directly.
+
+	if e.olap.Dialect() == drivers.DialectDruid {
+		dimensions := make([]Dimension, len(qry.Dimensions))
+		for i, d := range qry.Dimensions {
+			dimensions[i] = Dimension{Name: d}
+		}
+		q := &Query{
+			MetricsView: qry.MetricsView,
+			Dimensions:  dimensions,
+			Where:       qry.Where,
+			Having:      qry.Having,
+			TimeRange:   qry.TimeRange,
+			Limit:       qry.Limit,
+		}
+
+		if err := e.rewriteQueryTimeRanges(ctx, q, executionTime); err != nil {
+			return nil, err
+		}
+
+		ast, err := NewAST(e.metricsView, e.security, q, e.olap.Dialect())
+		if err != nil {
+			return nil, err
+		}
+
+		if err := e.rewriteLimitsIntoSubqueries(ast); err != nil {
+			return nil, err
+		}
+
+		// native search
+		res, err := qry.executeSearchInDruid(ctx, e.olap, ast)
+		if err == nil || !errors.Is(err, errDruidNativeSearchUnimplemented) {
+			return res, err
+		}
+	}
+
+	var (
+		finalSQL  strings.Builder
+		finalArgs []any
+		rowsCap   int64
+		err       error
+	)
+	for i, d := range qry.Dimensions {
+		if i > 0 {
+			finalSQL.WriteString(" UNION ALL ")
+		}
+		q := &Query{
+			MetricsView: qry.MetricsView,
+			Dimensions:  []Dimension{{Name: d}},
+			Having:      qry.Having,
+			TimeRange:   qry.TimeRange,
+			Limit:       qry.Limit,
+		}
+		q.Where = whereExprForSearch(qry.Where, d, qry.Search)
+
+		if err := e.rewriteQueryTimeRanges(ctx, q, executionTime); err != nil {
+			return nil, err
+		}
+
+		rowsCap, err = e.rewriteQueryEnforceCaps(q)
+		if err != nil {
+			return nil, err
+		}
+
+		ast, err := NewAST(e.metricsView, e.security, q, e.olap.Dialect())
+		if err != nil {
+			return nil, err
+		}
+
+		if err := e.rewriteLimitsIntoSubqueries(ast); err != nil {
+			return nil, err
+		}
+
+		sql, args, err := ast.SQL()
+		if err != nil {
+			return nil, err
+		}
+		finalSQL.WriteString(fmt.Sprintf("SELECT '%s' AS dimension, %s AS value FROM (%s)", d, e.olap.Dialect().EscapeIdentifier(d), sql))
```

Comment:
What if `d` contains a single quote? I think the dialect type might have a func for escaping a string, else we may need to add it (or put it in the args if possible to do at the right place)